x/auth/keeper/keeper.go

```diff
@@ -126,17 +135,21 @@ func NewAccountKeeper(
 	sb := collections.NewSchemaBuilder(storeService)
 
 	ak := AccountKeeper{
-		addressCodec:    ac,
-		bech32Prefix:    bech32Prefix,
-		storeService:    storeService,
-		proto:           proto,
-		cdc:             cdc,
-		permAddrs:       permAddrs,
-		authority:       authority,
-		Params:          collections.NewItem(sb, types.ParamsKey, "params", codec.CollValue[types.Params](cdc)),
-		AccountNumber:   collections.NewSequence(sb, types.GlobalAccountNumberKey, "account_number"),
-		Accounts:        collections.NewIndexedMap(sb, types.AddressStoreKeyPrefix, "accounts", sdk.AccAddressKey, codec.CollInterfaceValue[sdk.AccountI](cdc), NewAccountIndexes(sb)),
-		UnorderedNonces: collections.NewKeySet(sb, types.UnorderedNoncesKey, "unordered_nonces", collections.PairKeyCodec(collections.Int64Key, collections.BytesKey)),
+		addressCodec:         ac,
+		bech32Prefix:         bech32Prefix,
+		storeService:         storeService,
+		proto:                proto,
+		cdc:                  cdc,
+		permAddrs:            permAddrs,
+		authority:            authority,
+		Params:               collections.NewItem(sb, types.ParamsKey, "params", codec.CollValue[types.Params](cdc)),
+		AccountNumber:        collections.NewSequence(sb, types.GlobalAccountNumberKey, "account_number"),
+		Accounts:             collections.NewIndexedMap(sb, types.AddressStoreKeyPrefix, "accounts", sdk.AccAddressKey, codec.CollInterfaceValue[sdk.AccountI](cdc), NewAccountIndexes(sb)),
+		AccountIDByAddress:   collections.NewMap(sb, types.AccountIDByAddressPrefix, "account_id_by_address", collections.PairKeyCodec(collections.Uint16Key, collections.BytesKey), collections.BytesValue),
+		AddressByAccountID:   collections.NewMap(sb, types.AddressByAccountIDPrefix, "address_by_account_id", collections.PairKeyCodec(collections.BytesKey, collections.Uint16Key), collections.BytesValue),
+		UnorderedNonces:      collections.NewKeySet(sb, types.UnorderedNoncesKey, "unordered_nonces", collections.PairKeyCodec(collections.Int64Key, collections.BytesKey)),
+		addressSpaceManagers: map[AddressSpacePrefix]AddressSpaceManager{},
+		addressPrefixByName:  map[string]AddressSpacePrefix{},
```

Comment:
how will these maps be populated given that they are in memory

Reply:
With `AccountKeeper.DefineAddressSpace` from `app.go`